LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceClient.cs

```diff
@@ -155,12 +154,165 @@ public async Task<bool> SendEventAsync(LoRaDeviceTelemetry telemetry, Dictionary
             return false;
         }
 
+        readonly SemaphoreSlim receiveAsyncLock = new SemaphoreSlim(1, 1);
```

Comment:
SemaphoreSlim has a Dispose and that should be called to cleanup unmanaged resources and avoid finalization.